container/shim/src/index.js

```diff
@@ -42,161 +47,231 @@ const PROXY_RESPONSE_HEADERS = [
 
 const argv = parseArgs(process.argv.slice(2))
 
-if (cluster.isPrimary) {
-  debug('Saturn L1 Node')
-  debug.extend('id')(nodeId)
-  debug.extend('version')(NODE_VERSION)
-  debug.extend('important')('===== IMPORTANT =====')
-  debug.extend('important')(`Earnings will be sent to Filecoin wallet address: ${FIL_WALLET_ADDRESS}`)
-  debug.extend('important')(NODE_OPERATOR_EMAIL ? `Payment notifications and important update will be sent to: ${NODE_OPERATOR_EMAIL}` : 'NO OPERATOR EMAIL SET, WE HIGHLY RECOMMEND SETTING ONE')
-  debug.extend('important')('===== IMPORTANT =====')
-
-  for (let i = 0; i < cpus().length; i++) {
-    cluster.fork()
+debug('Saturn L1 Node')
+debug.extend('id')(nodeId)
+debug.extend('version')(NODE_VERSION)
+debug.extend('important')('===== IMPORTANT =====')
+debug.extend('important')(`Earnings will be sent to Filecoin wallet address: ${FIL_WALLET_ADDRESS}`)
+debug.extend('important')(NODE_OPERATOR_EMAIL ? `Payment notifications and important update will be sent to: ${NODE_OPERATOR_EMAIL}` : 'NO OPERATOR EMAIL SET, WE HIGHLY RECOMMEND SETTING ONE')
+debug.extend('important')('===== IMPORTANT =====')
+
+process.on('SIGQUIT', shutdown)
+process.on('SIGINT', shutdown)
+
+setTimeout(async function () {
+  if (argv.register !== false) {
+    await register(true).catch(err => {
+      debug(`Failed to register ${err.name} ${err.message}`)
+      process.exit(1)
+    })
   }
 
-  cluster.on('exit', () => {
-    if (Object.keys(cluster.workers).length === 0) {
-      debug('All servers closed')
-      shutdownCluster()
-    }
-  })
+  // Start log ingestor
+  await initLogIngestor()
+}, 500)
 
-  process.on('SIGQUIT', shutdownCluster)
-  process.on('SIGINT', shutdownCluster)
+const ipfsAgent = new https.Agent({
+  keepAlive: true,
+  maxSockets: Math.floor(256 / cpus().length)
+})
 
-  setTimeout(async function () {
-    if (argv.register !== false) {
-      await register(true).catch(err => {
-        debug(`Failed to register ${err.name} ${err.message}`)
-        process.exit(1)
-      })
-    }
+const app = express()
```

Comment:
Express is known to have a pretty bad performance, plus it's not very actively maintained. I strongly recommend replacing it with a more modern alternative like  [Fastify](https://www.fastify.io).

Having said that, such change is most likely out of scope of this pull request.

Reply:
I agree, we have an open issue for this: https://github.com/filecoin-saturn/L1-node/issues/33